### PR TITLE
[WIP] Remove global step update in training type plugins

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Unblocks #8598

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/deepspeed.py [hunk 1]
```diff
@@ -780,13 +783,6 @@ def load_optimizer_state_dict(self, checkpoint: Mapping[str, Any]) -> None:
         # override to do nothing, deepspeed engine already loaded the states in `load_checkpoint_file()`
         pass
 
-    def update_global_step(self, total_batch_idx: int, current_global_step: int) -> int:
```
Comment (tchaton) [inline pytorch_lightning/plugins/training_type/deepspeed.py:783]: Why did you remove this ? DeepSpeed doesn't perform an update on latest batch of an epoch as Lightning currently does.

Comment (tchaton) [reply pytorch_lightning/plugins/training_type/deepspeed.py:783]: This is why we introduce `update_global_step` hook.
